### PR TITLE
Avoid division by zero

### PR DESCRIPTION
**Pull request by hitottiez**
In Contrastive.backward(), division by zero occurs when ```dist``` is equal to zero.

```
gx0 += alpha * (1 - y) * mdist_p * mdist * -(self.diff / dist)
```

It causes that gradient value is nan.

Please see my proposed modification.

### CODE REVIEW

**Comment by okuta** [comment]
Please add test case what ``dist`` is zero.

**Comment by okuta** [comment]
Commits are strange.
Would you organize commits?

**Comment by hitottiez** [comment]
I added test cases as follows.
e0e95e5cb2aac451f62510d3e0f0c8e3b66cf383

File: tests/chainer_tests/functions_tests/loss_tests/test_contrastive.py [hunk 1]
```diff
@@ -80,5 +80,20 @@ def test_backward_gpu_no_cudnn(self):
         self.check_backward(cuda.to_gpu(self.x0), cuda.to_gpu(self.x1),
                             cuda.to_gpu(self.t))
 
+    @condition.retry(3)
+    def test_backward_zero_dist_cpu(self):
+        x0 = self.x0
+        x1 = self.x0
+        self.dist = xp.sqrt(numpy.sum((x0 - x1) ** 2, axis=1))
+        self.check_backward(x0, x1, self.t)
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/loss_tests/test_contrastive.py]: How about `self.check_backward(self.x0, self.x0, self.t)`?

File: tests/chainer_tests/functions_tests/loss_tests/test_contrastive.py [hunk 2]
```diff
@@ -80,5 +80,20 @@ def test_backward_gpu_no_cudnn(self):
...
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_backward_zero_dist_gpu_no_cudnn(self):
+        x0 = self.x0
+        x1 = self.x0
+        self.dist = cuda.to_gpu(numpy.sqrt(numpy.sum((x0 - x1) ** 2, axis=1)))
+        self.check_backward(cuda.to_gpu(x0), cuda.to_gpu(x1),
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/loss_tests/test_contrastive.py]: ditto

**Comment by hitottiez** [comment]
Thank you for your suggestion.

I modified test cases as follows.
ebbfa7b33b314715e43028c630cbfa7816301b54

**Comment by okuta** [comment]
LGTM
